src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java

```diff
@@ -1201,8 +1201,11 @@ public Image getIcon(int width, int height) {
      */
     static Image getSystemIcon(SystemIcon iconType) {
         long hIcon = getSystemIcon(iconType.getIconID());
+        if (hIcon == 0) {
+            return null;
+        }
         Image icon = makeIcon(hIcon);
```

Comment:
```suggestion
        }

        Image icon = makeIcon(hIcon);
```
I'd add a blank line here to visually separate the two blocks: deal with `null`, then handle the “happy” case.